modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/DLCMessageExpiryTestCase.java

```diff
@@ -0,0 +1,180 @@
+import org.wso2.mb.integration.common.clients.AndesClient;
+import org.wso2.mb.integration.common.clients.configurations.AndesJMSConsumerClientConfiguration;
+import org.wso2.mb.integration.common.clients.configurations.AndesJMSPublisherClientConfiguration;
+import org.wso2.mb.integration.common.clients.exceptions.AndesClientConfigurationException;
+import org.wso2.mb.integration.common.clients.exceptions.AndesClientException;
+import org.wso2.mb.integration.common.clients.operations.clients.AndesAdminClient;
+import org.wso2.mb.integration.common.clients.operations.utils.ExchangeType;
+import org.wso2.mb.integration.common.clients.operations.utils.JMSAcknowledgeMode;
+import org.wso2.mb.integration.common.utils.backend.ConfigurationEditor;
+import org.wso2.mb.integration.common.utils.backend.MBIntegrationBaseTest;
+import org.xml.sax.SAXException;
+
+import javax.jms.JMSException;
+import javax.naming.NamingException;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.rmi.RemoteException;
+
+/**
+ * This class includes test cases to test expired message deletion in DLC
+ */
+public class DLCMessageExpiryTestCase extends MBIntegrationBaseTest {
+    /**
+     * Test queue name
+     */
+    private static final String TEST_QUEUE_DLC_EXPIRY = "DLCTestQueue";
+
+     /**
+     * Initializes test case
+     *
+     * @throws XPathExpressionException
+     */
+    @BeforeClass()
+    public void init() throws XPathExpressionException, MalformedURLException {
+        super.init(TestUserMode.SUPER_TENANT_USER);
+    }
+
+    /**
+     * Set topicMessageDeliveryStrategy to DISCARD_ALLOWED so that broker will simulate an acknowledgement
+     * if some subscribers are slow to acknowledge the message
+     *
+     * @throws XPathExpressionException
+     * @throws IOException
+     * @throws ConfigurationException
+     * @throws SAXException
+     * @throws XMLStreamException
+     * @throws LoginAuthenticationExceptionException
+     * @throws URISyntaxException
+     * @throws AutomationUtilException
+     */
+    @BeforeClass
+    public void setupConfiguration() throws XPathExpressionException, IOException,
+            ConfigurationException, SAXException, XMLStreamException, LoginAuthenticationExceptionException,
+            URISyntaxException, AutomationUtilException {
+        super.serverManager = new ServerConfigurationManager(automationContext);
+        String defaultMBConfigurationPath = ServerConfigurationManager.getCarbonHome() +
+                File.separator + "repository" + File.separator + "conf" + File.separator + "broker.xml";
+        ConfigurationEditor configurationEditor = new ConfigurationEditor(defaultMBConfigurationPath);
+
+        configurationEditor.updateProperty(AndesConfiguration.PERFORMANCE_TUNING_EXPIRE_MESSAGES_IN_DLC, "true");
```

Comment:
add a afterClass method and revert the configuration change done in afterClass method